lib/intercom/service/conversation.rb

```diff
@@ -61,6 +61,28 @@ def run_assignment_rules(id)
         response = @client.post("/#{collection_name}/#{id}/run_assignment_rules")
         collection_class.new.from_response(response)
       end
+
+      def redact(redact_data)
+        redact_type = redact_data.fetch(:type) { raise 'type field is required' }
+        conversation_id = redact_data.delete(:id) { raise 'id field is required' }
+        params = {
+            type: redact_type,
+            conversation_id: conversation_id
+        }
+
+        if redact_type == 'source'
+          source_id = redact_data.fetch(:source_id) { raise 'source_id field is required for type "source"' }
+          params[:source_id] = source_id
+        elsif redact_type == 'conversation_part'
+          conversation_part_id = redact_data.fetch(:conversation_part_id) { raise 'conversation_part_id field is required for type "conversation_part"' }
+          params[:conversation_part_id] = conversation_part_id
+        else
+          raise 'unknown redact_type'
+        end
+
+        response = @client.post("/#{collection_name}/redact", params)
```

Comment:
Like the suggestion of introducing some idempotency in the API.

Meanwhile, we can handle the error here by adding `conversation_part_not_redactable` to L195
https://github.com/intercom/intercom-ruby/blob/1aa86b04698a497aeea2f580c3ed1a9fa3539b1d/lib/intercom/request.rb#L195-L196

Should result in a,
```
Intercom::BadRequestError: Conversation part can't be redacted

Reply:
Can you share some info why some parts cannot be redacted? I've redacted a lot of conversations parts and it seems kind of random being both sources and conversation parts. And how to solve it?

Reply:
As far as I know,
- We've got some info in the docs [here](https://developers.intercom.com/intercom-api-reference/reference#redact-a-conversation-part) outlining which conversation parts can be redacted.
- Then there's the case of trying to redact an already redacted source/part which would result in such an error. 
- And source messages started via articles are  currently not redactable.

Will check with the team and see if there are any more cases, and update the docs too to reflect the same.